### PR TITLE
TE-269 Icon

### PR DESCRIPTION
**Pull request by marcdavi-es**
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-269)

### What **one** thing does this PR do?
Creates component, examples and tests for Icon.

screen shot 2018-04-03 at 16 35 35

### CODE REVIEW

File: src/components/elements/Icon/component.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import { Paragraph } from 'typography/Paragraph';
+
+/**
+ * An icon is a glyph used to represent something else.
+ * @return {Object}
+ */
+export const Component = ({ isDisabled, label, name }) => (
+  <div>
+    <Icon
+      color={isDisabled ? 'grey' : undefined}
```
Comment (davidpelayo) [inline src/components/elements/Icon/component.js]: Props should never setup color literals at a render level?
I know there is a enum with colors defined in Semantic, and that `inverted` controls how to contrast colors.
Maybe we should customize always via CSS and use mixins for contrast colors?

File: src/components/elements/Icon/component.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      inverted={isDisabled}
+      name={name}
+      size="large"
```
Comment (davidpelayo) [inline src/components/elements/Icon/component.js]: This should exposed via props maybe, falling back to `large`?

File: src/components/elements/Icon/component.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import { Icon } from 'semantic-ui-react';
+
+import { Paragraph } from 'typography/Paragraph';
+
+/**
+ * An icon is a glyph used to represent something else.
+ * @return {Object}
+ */
+export const Component = ({ isDisabled, label, name }) => (
+  <div>
```
Comment (davidpelayo) [inline src/components/elements/Icon/component.js]: Icons should be always render inline, I think. Using `<span>` maybe?

File: src/components/elements/Icon/component.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    <Icon
+      color={isDisabled ? 'grey' : undefined}
+      inverted={isDisabled}
+      name={name}
+      size="large"
+    />
+    {label && <Paragraph>{label}</Paragraph>}
```
Comment (davidpelayo) [inline src/components/elements/Icon/component.js]: From the accessibility point of view, we should only use `<label>Insert icon here</label>` when a label is informed. Maybe we do not want to go there now?

File: src/components/elements/Icon/component.js (hunk 1 above)
Comment (marcdavi-es) [reply src/components/elements/Icon/component.js]: Here I'm just trying to use the existing API for Icon to get us where we need to be. This requires zero extra CSS.

The alternative is setting up a new custom class (`is-disabled` or whatever) and then writing extra CSS to get us to the same place.

I'm always in favour of less code...?

File: src/components/elements/Icon/component.js (hunk 2 above)
Comment (marcdavi-es) [reply src/components/elements/Icon/component.js]: I think for now I'm happy limiting this to one default. We can easily extend it later.

File: src/components/elements/Icon/component.js (hunk 1 above)
Comment (davidpelayo) [reply src/components/elements/Icon/component.js]: If we think we can live with that, totally agree. If it's a requirement to customize colors for inline content (including icons) - then we need it.
I haven't reviewed the Livingstone template layouts, but we have had cases were inline content (text and icons) needed to be restyled and coloured concretely given the user selected theme.

File: src/components/elements/Icon/component.js (hunk 4 above)
Comment (davidpelayo) [reply src/components/elements/Icon/component.js]: Agreed.

File: src/components/elements/Icon/component.js (hunk 1 above)
Comment (marcdavi-es) [reply src/components/elements/Icon/component.js]: Yeah for now I think we just want it hardcoded :)

File: src/components/elements/Icon/component.js (hunk 3 above)
Comment (marcdavi-es) [reply src/components/elements/Icon/component.js]: Changed

**Pull request merged**